src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -28,16 +32,30 @@ public HttpHandlerResourceV3Provider()
 
             HttpHandlerResourceV3 curResource = null;
 
-            var clientHandler = CreateCredentialHandler(source.PackageSource);
+            if (source.PackageSource.IsHttp)
```

Comment:
shouldn't we throw instead? Otherwise we are just going to swallow bad scenarios